server/src/test/java/org/elasticsearch/index/mapper/DocCountFieldMapperTests.java

```diff
@@ -57,4 +71,44 @@ public void testInvalidDocument_ArrayDocCount() throws Exception {
         Exception e = expectThrows(MapperParsingException.class, () -> mapper.parse(source(b -> b.array(CONTENT_TYPE, 10, 20, 30))));
         assertThat(e.getCause().getMessage(), containsString("Arrays are not allowed for field [_doc_count]."));
     }
+
+    public void testSyntheticSource() throws IOException {
+        DocumentMapper mapper = createDocumentMapper(syntheticSourceMapping(b -> {}));
+        assertThat(syntheticSource(mapper, b -> b.field(CONTENT_TYPE, 10)), equalTo("{\"_doc_count\":10}"));
+    }
+
+    public void testSyntheticSourceMany() throws IOException {
+        DocumentMapper mapper = createDocumentMapper(syntheticSourceMapping(b -> {}));
+        List<Integer> counts = randomList(2, 10000, () -> between(1, Integer.MAX_VALUE));
+        try (Directory directory = newDirectory()) {
```

Comment:
Does `withLuceneIndex` not work here?

Reply:
Let me have another look. I think there was a reason, but it might not apply right here.